### PR TITLE
re-organize repo to make more sense

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
changes:
 - lib => build
 - top-level files have moved to `src`
 - converted to flat monorepo... flanorepo? to have the tests be their own package so that they can test imports/exports
 
 
 **Test Plan**
  - locally, `pnpm build` should give no diff
  - locally, run `FIGMA_TOKEN=<your personal token> pnpm figma:export-styles`, should give no diff. the IDs for the figma documents can be found in our Toucan documentation app's README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ pnpm run build:preview
 npx http-server ./dist
 ```
 
-Note that changes to config or build scripts will require re-running `pnpm start`.
+Note that changes to `src` or `build` scripts will require re-running `pnpm start`.
```
Comment (nicolechung) [inline README.md:60]: Question: isn't the point to NOT have to rerun `pnpm start` when changing a `src` file?

**Comment by nicolechung** [comment]
I am getting an error when running `pnpm figma:export-styles` about being unable to find a local issuer certificate

```
✖ fetching document
Error: while fetching file "undefined": unable to get local issuer certificate
    at /Users/mystuff/tailwind-toucan-base/node_modules/@figma-export/core/dist/lib/export-styles.js:22:15
```

File: README.md (hunk 1 above)
Comment (NullVoxPopuli) [reply README.md:60]: not the point of this PR -- `pnpm start` doesn't have any "watch mode" implemented -- and I'm not sure how to implement it -- maybe I could switch everything to rollup plugins? but idk -- problem for another time

**Comment by NullVoxPopuli** [comment]
So, we found out something accidentally happened to a Figma page -- but it has since been restored :sweat_smile:

**Pull request merged**